prototype2/multiblade/mbcaen/DataParser.cpp

```diff
@@ -0,0 +1,48 @@
+/** Copyright (C) 2017-2018 European Spallation Source */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Class to parse detector readout for Multi-Blade using Jadaq UDP
+/// Data format is described in
+/// https://github.com/ess-dmsc/jadaq/blob/devel/DataFormat.hpp
+///
+//===----------------------------------------------------------------------===//
+
+#include <common/Trace.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <mbcaen/DataParser.h>
+#include <memory>
+
+#undef TRC_LEVEL
+#define TRC_LEVEL TRC_L_DEB
+
+int DataParser::parse(const char *buffer, unsigned int size) {
+
+  MBHeader = nullptr;
+  MBData = nullptr;
+  memset(&Stats, 0, sizeof(struct Stats));
```

Comment:
Not sure if this is the best way of doing it? Maybe it is, but I'm just wondering if there is a more C++ way of doing it.

Reply:
Fair enough, I use memset with simple structs containing numbers and counters that need to be cleared. I find it easy to see what it is doing and what the intent is. It is certainly the lest C++ way :-)

Reply:
This question was interesting enough that I had to do a search for a potential solution. Using `memset()` works if your struct contains only primitive data types (e.g. `int`) but will lead to a crash or undefined behaviour if it contains non-primitive types.

It appears that the C++ (and thus safe) way of doing this is by using the "default initialiser" `{}`. I.e.:
```
Stats = {};
```
As far as I can discern, this will set the value of primitive members to 0 and invoke the default initialiser of non-primitive members.